Makefile

```diff
@@ -0,0 +1,31 @@
+.PHONY: help
+help:
+	@echo 'Usage:'
+	@echo '  make <target>'
+	@echo ''
+	@echo 'Targets:'
+	@awk 'match($$0, /^([a-zA-Z_\/-]+):.*?## (.*)$$/, m) {printf "  \033[36m%-30s\033[0m %s\n", m[1], m[2]}' $(MAKEFILE_LIST) | sort
+
+CONTAINERFILE=Containerfile
+CONTAINER_IMAGE=osbuild-images
+CONTAINER_EXECUTABLE?=podman
+
+BASE_CONTAINER_IMAGE_NAME?=registry.fedoraproject.org/fedora
+BASE_CONTAINER_IMAGE_TAG?=41
+BASE_CONTAINER_IMAGE?=${BASE_CONTAINER_IMAGE_NAME}:${BASE_CONTAINER_IMAGE_TAG}
+
+container_built.info: $(CONTAINERFILE) Schutzfile test/ go.mod go.sum # internal rule to build the container only if needed
+	$(CONTAINER_EXECUTABLE) build --build-arg BASE_CONTAINER_IMAGE="${BASE_CONTAINER_IMAGE}" \
+	                              --tag $(CONTAINER_IMAGE) \
+	                              -f $(CONTAINERFILE) .
+	echo "Container last built on" > $@
+	date >> $@
+
+.PHONY: test
+test: container_built.info ## run all tests
```

Comment:
I would be surprised if `make test` runs a container for me, maybe that is just me but I would expect more something like `go test` here. Maybe just a different makefile target (`make test-in-container`, `gh-testss`, `test-gh` or something)?

And/or two targets: "test" that just runs the test locally (which is I think what most ppl expect) and test-gh or something that puts them into the container?

Reply:
In general you are right. I think this is more something we can enrich at any point in time, not to enlarge this PR even more.
e.g. rename the implementation of this PR to a `gh-test` target and implement a `test` target without a container.

For me the most important thing is that `make test` just runs the tests and they should behave the same as the github tests as close as possible. (That was the whole plan of this issue)

I'd postpone the implementation of the second "test target" (testing without the container) to the future if we happen to need it.

Reply:
Sure, let's keep this PR targeted. I still feel a `make test` that creates/runs a container is not what I (and I assume most users) would expect. Calling it `make gh-test` would still be my suggestion. I am okay to be overridden by the more experienced images developers like @achilleas-k, @thozza or @ondrejbudai here of course.

Reply:
for convenience are you ok with renaming it to `make gh-test` but still have a
```
# TBD implement checking/installing prerequisites and running the tests without a container
test: gh-test
```
and I'll check for a real `make test` after this PR (not to enrich this one)

Reply:
… or even name it `test_in_container` or similar

Reply:
+1 for naming it `gh-test` or `gh-action-test` (indicating that it's replicating the action, not actually triggering a test on github or something).  `test_in_container` also works, but is an awkward name for a make target IMO.



> for convenience are you ok with renaming it to `make gh-test` but still have a
> 
> ```
> # TBD implement checking/installing prerequisites and running the tests without a container
> test: gh-test
> ```
> 
> and I'll check for a real `make test` after this PR (not to enrich this one)

No, I'd rather `make test` just did
```
go test -race ./...
go test -race ./pkg/dnfjson/... -force-dnf
```

and have it fail on systems that don't have dependencies.

Reply:
please check the revised implementation and documentation

Reply:
I agree with `make test` to just running tests locally and not in a container. One of the use cases may be to intentionally running them in a git checkout on a different distro, such as CentOS Stream or RHEL.